server/app/services/languageProcessor/TaskMap.java

```diff
@@ -14,37 +20,69 @@
    * @throws InvocationTargetException
    * @throws IllegalAccessException
    */
-    public static String questionMapping(String methodName, String argName) throws NoSuchMethodException,
-            InvocationTargetException,IllegalAccessException{
+  final static String ERROR_MSG = "Sorry, I cannot understand the question";
+
+  public static JsonNode questionMapping(String methodName, String argName, JsonNode responseBody){
+
+    TaskMap taskMap = new TaskMap();
 
-        TaskMap tm = new TaskMap();
+    try {
 
-        //call the method at runtime according to the argument "methodName"
-        Method m = TaskMap.class.getMethod(methodName, String.class);
-        String returnVal = (String) m.invoke(tm, argName);
-        return returnVal;
+      //call the method at runtime according to the argument "methodName"
+      Method m = TaskMap.class.getMethod(methodName, String.class, JsonNode.class);
+      JsonNode returnVal = (JsonNode) m.invoke(taskMap, argName, responseBody);
+      return returnVal;
+
+    }catch (NoSuchMethodException e){
+      e.printStackTrace();
+      return taskMap.parseToJson("fail",ERROR_MSG);
+    }catch (InvocationTargetException e){
+      e.printStackTrace();
+      return taskMap.parseToJson("fail",ERROR_MSG);
+    }catch (IllegalAccessException e){
+      e.printStackTrace();
+      return taskMap.parseToJson("fail",ERROR_MSG);
+    }catch (NullPointerException e){
+      return taskMap.parseToJson("fail",ERROR_MSG);
     }
+  }
 
   /**
    * This method requests ticket info and returns it to the calling method
    * @param ticket
    * @return
    */
-    public String description_of_ticket(String ticket){
-        String answer = "Description of the ticket " + ticket;
-        System.out.println(answer);
-        //String ans = JIRAconnector.getDescription(ticket);
-        return answer;
-    }
+  public JsonNode description_of_ticket(String ticket, JsonNode responseBody){
```

Comment:
Instead of `ticket` it should be named as `issue` as that's what JIRA calls it officially.